pkg/config/polling_manager.go

```diff
@@ -215,6 +217,13 @@ func (cm *PollingProjectConfigManager) GetConfig() (ProjectConfig, error) {
 	return cm.projectConfig, nil
 }
 
+// GetConfig returns the project config
+func (cm *PollingProjectConfigManager) GetOptimizelyConfig() *OptimizelyConfig {
+	cm.configLock.RLock()
+	defer cm.configLock.RUnlock()
+	return cm.optimizelyConfig
```

Comment:
Here we lazily create the optimizelyConfig so that we only ever take up the memory if this method is used.

Reply:
this always returns either null or populated optimizely config. it always takes up the memory to construct it every time a datafile is updated.

Reply:
ok, we have clarified over IM